File: Practice/AllDomains/CoreCS/Algorithms/Strings/MarsExploration.cs
/*	author:@shivkrthakur  */
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
class Solution 
{
    static void Main(String[] args) 
	{
        string s = Console.ReadLine();
        string sos = "SOS";
        int count = 0;
        for(int i = 0; i < s.Length; i++)
        {
            if(s[i] != sos[i%3]) count++;
        }
        Console.WriteLine(count);
    }
}

File: Practice/AllDomains/CoreCS/Algorithms/Strings/MarsExploration.js
/*	author:@shivkrthakur  */
process.stdin.resume();
process.stdin.setEncoding('ascii');

var input_stdin = "";
var input_stdin_array = "";
var input_currentline = 0;

process.stdin.on('data', function (data) {
    input_stdin += data;
});

process.stdin.on('end', function () {
    input_stdin_array = input_stdin.split("\n");
    main();    
});

function readLine() {
    return input_stdin_array[input_currentline++];
}

/////////////// ignore above this line ////////////////////

function main() {
    var S = readLine();
    var sos = "SOS";
    var count = 0;
    for(var i = 0; i < S.length; i++){
        if(S[i] != sos[i % 3]) count++;
    }
    console.log(count);
}
